### PR TITLE
Pf4 build output

### PR DESCRIPTION
**Pull request by Hyperkid123**

### CODE REVIEW

File: packages/pf4-component-mapper/rollup.config.js [hunk 1]
```diff
@@ -4,66 +4,103 @@ import babel from 'rollup-plugin-babel';
...
+  terser({
+    keep_classnames: true,
+    keep_fnames: true
+  }),
+  sass({
+    insert: true
+  })
+];
+
+// export default [
```
Comment (rvsia) [inline packages/pf4-component-mapper/rollup.config.js]: some commented code

File: packages/pf4-component-mapper/rollup.config.js [hunk 2]
```diff
@@ -4,66 +4,103 @@ import babel from 'rollup-plugin-babel';
...
-  'prop-types',
-  '@data-driven-forms/react-form-renderer',
-  '@patternfly/react-core/**',
-  '@patternfly/react-icons/**',
-], null, { resolve: false });
+import glob from 'glob';
+import path from 'path';
+
+const outputPaths = glob.sync(path.resolve(__dirname, './src/components/*.js'));
+console.log('outputPaths', outputPaths);
```
Comment (rvsia) [inline packages/pf4-component-mapper/rollup.config.js]: console.log

File: packages/pf4-component-mapper/src/components/text-area.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  <FormGroup
+    label={label}
+    isRequired={isRequired}
+    helperText={helperText}
+    meta={meta}
+    description={description}
+    hideLabel={hideLabel}
+    id={id || input.name}
+  >
+    <Pf4TextArea disabled={isDisabled || isReadOnly} {...input} {...rest} />;
```
Comment (rvsia) [inline packages/pf4-component-mapper/src/components/text-area.js]: `;` here

File: packages/pf4-component-mapper/src/components/text-field.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import FormGroup from '../common/form-group';
+import { TextInput } from '@patternfly/react-core/dist/js/components/TextInput/TextInput';
+import PropTypes from 'prop-types';
+
+const TextField = ({ label, isRequired, helperText, meta, description, hideLabel, input, isReadOnly, isDisabled, id, ...rest }) => {
```
Comment (rvsia) [inline packages/pf4-component-mapper/src/components/text-field.js]: unnecessary block

File: packages/pf4-component-mapper/src/components/switch.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  id,
+  ...rest
+}) => (
+  <FormGroup
+    label={label}
+    isRequired={isRequired}
+    helperText={helperText}
+    meta={meta}
+    description={description}
+    hideLabel={hideLabel}
```
Comment (rvsia) [inline packages/pf4-component-mapper/src/components/switch.js]: Shouldn't be always true?

File: packages/pf4-component-mapper/src/components/checkbox.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import { Checkbox as Pf4Checkbox } from '@patternfly/react-core/dist/js/components/Checkbox/Checkbox';
+
+const SingleCheckbox = ({ label, isRequired, helperText, meta, description, hideLabel, input, isReadOnly, isDisabled, id, ...rest }) => (
+  <FormGroup
+    label={label}
+    isRequired={isRequired}
+    helperText={helperText}
+    meta={meta}
+    description={description}
+    hideLabel={hideLabel}
```
Comment (rvsia) [inline packages/pf4-component-mapper/src/components/checkbox.js]: Shouldn't be always true?

File: packages/pf4-component-mapper/src/components/checkbox.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    id={id || input.name}
+  >
+    <Pf4Checkbox
+      isChecked={input.checked}
```
Comment (rvsia) [inline packages/pf4-component-mapper/src/components/checkbox.js]: it should also get `id={id || input.name}`

File: packages/pf4-component-mapper/src/components/checkbox.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+      {...input}
+      label={rest.title || rest.label}
```
Comment (rvsia) [inline packages/pf4-component-mapper/src/components/checkbox.js]: label is not in rest

File: packages/pf4-component-mapper/src/common/multiple-choice-list.js [hunk 9]
```diff
@@ -6,37 +6,22 @@ import { Checkbox } from '@patternfly/react-core/dist/js/components/Checkbox/Che
...
-    label={ label }
-    fieldId={ rest.name }
-    isValid={ !showError }
-    isRequired={ isRequired }
-    helperText={ helperText }
-    helperTextInvalid={ meta.error }
-  >
-    { children }
+const Wrapper = ({ showError, isRequired, helperText, label, meta, children, rest }) => (
+  <FormGroup label={label} fieldId={rest.name} isValid={!showError} isRequired={isRequired} helperText={helperText} helperTextInvalid={meta.error}>
```
Comment (rvsia) [inline packages/pf4-component-mapper/src/common/multiple-choice-list.js]: Please use the new commont FormGroup

**Comment by rvsia** [comment]
Multiple checkbox is not changed by clicking on labels

**Comment by rvsia** [comment]
Radio does not show validation errors

File: packages/pf4-component-mapper/src/common/multiple-choice-list.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+import FormGroup from './form-group';
+
+const FinalCheckbox = (props) => <Checkbox isChecked={props.checked} {...props} />;
+
+FinalCheckbox.propTypes = {
+  checked: PropTypes.bool
+};
+
+const Wrapper = ({ isRequired, helperText, label, meta, children, rest }) => (
+  <FormGroup label={label} meta={meta} fieldId={rest.name} isRequired={isRequired} helperText={helperText}>
```
Comment (rvsia) [inline packages/pf4-component-mapper/src/common/multiple-choice-list.js]: You are using just `id` instead of `fieldId`

**Comment by rvsia** [comment]
errors appear:

```
checkPropTypes.js:20 Warning: Failed prop type: The prop `componentType` is marked as required in `FormGroup`, but its value is `undefined`.
    in FormGroup (created by TextArea)
    in TextArea (created by Field)
    in Field (created by Context.Consumer)
    in ReactFinalForm(Field) (created by FieldProvider)
    in FieldProvider (created by FieldWrapper)
    in FieldWrapper (created by Context.Consumer)
    in FormFieldHideWrapper (created by Context.Consumer)
    in FormConditionWrapper (created by Context.Consumer
```

```
TextArea: TextArea requires either an id or aria-label to be specified
```

```
react-dom.development.js:530 Warning: React does not recognize the `FieldProvider` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `fieldprovider` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    in textarea (created by TextArea)
    in TextArea (created by TextArea)
```

```
react-dom.development.js:530 Warning: React does not recognize the `FieldArrayProvider` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `fieldarrayprovider` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    in textarea (created by TextArea)
    in TextArea (created by TextArea)
```

```
react-dom.development.js:530 Warning: React does not recognize the `formOptions` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `formoptions` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    in textarea (created by TextArea)
    in TextArea (created by TextArea)
```

```
Warning: Failed prop type: Invalid prop `hideLabel` of type `boolean` supplied to `TextField`, expected `string`.
    in TextField (created by Field)
    in Field (created by Context.Consumer)
```

```
backend.js:6 Warning: Failed prop type: Invalid prop `options[0].value` of type `number` supplied to `Checkbox`, expected `string`.
    in Checkbox (created by Field)
    in Field (created by Context.Consumer)
    in ReactFinalForm(Field) (created by FieldProvider)
    in FieldProvider (created by FieldWrapper)
```

```
backend.js:6 Warning: React does not recognize the `FormSpyProvider` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `formspyprovider` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    in div (created by Tabs)
    in Tabs (created by Context.Consumer)
    in ComponentWithOuia (created by Context.Consumer)
    in Unknown (created by FormTabs)
    in FormTabs (created by FieldWrapper)
```

**Comment by Hyperkid123** [comment]
@rvsia i want to resolve these errors globally with some innerProps, removal of the form options etc. I added all these props back to ensure we won't miss anything once we start dealing with it.

I would like to remove the responsibility to clean the props before implementing component in mapper. Basically just do something like and have no errors.
```jsx
import FieldProvider from '@data..../FieldProvider';
export const TextField = props => <FieldProvider component={SomeComponent} {...props} />
```

**Comment by rvsia** [comment]
@Hyperkid123 Yeah, but not all of these errors are caused by shared uncatched props.

**Comment by Hyperkid123** [comment]
@rvsia yup. Eventually, before v2 is finished, they should exist and should be accessible either via innerProps in props or via context. 90% of the time you don't even touch them. I would like to separate Renderer props from the actual field props.

**Pull request merged**